internal/pkg/scaffold/cmd.go

```diff
@@ -159,4 +164,14 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func serveCRMetrics(cfg *rest.Config) {
+	operatorSpecificScheme := kuberuntime.NewScheme()
+	apis.AddToScheme(operatorSpecificScheme)
+	filteredScheme := kubemetrics.FilterOutMetaTypes(operatorSpecificScheme)
```

Comment:
I choose to put the filtering in the users main.go file because we make a lot of assumptions when filtering out the types, so in case we get it wrong the users can do custom filtering or append some types we filter out initially.

Reply:
Hmm. I like the idea of auto-discovering the GVKs we add to the scheme, but I'm worried that this could be a whack-a-mole situation where other meta types are added in later versions of Kubernetes and we end up having to play catch up (or forcing our users to).

In `FilterOutMetaTypes()`, is it possible to get the list of meta types from a Kubernetes library, rather than hardcoding it?

Reply:
> In FilterOutMetaTypes(), is it possible to get the list of meta types from a Kubernetes library, rather than hardcoding it?

I agree, I don't like it either, but had a look at the codebase and it seems like they register it one by one, not from a known public list. There will be an error in our CI if they add any new types later one when we switch to new k8s version, and we can add them to the list. I suggest we leave it like this, and since this code is not part of the users codebase we can modify in the future if we find a better solution, WDYT?
 https://github.com/kubernetes/apimachinery/blob/a44ef629a3c96737242430c412751bca67b342bd/pkg/apis/meta/internalversion/register.go#L56

Reply:
> I didn't get that far yet in the code review. That certainly helps. 👍

Yes, the errors are produced when we try to generate the metrics as the GVK does not exist for this custom resource, so we will most likely catch this before our users do. 

Also I am not sure upstream actually plans on adding any new meta types in the future. Also the users operator will never erorr out just the error will be logged, so it won't effect them in case that happens.

> this method vs. the explicit method of building up the metrics registration similar to the APIs scheme building pattern.
Personally, I think this is less code to the user and with that less complexity. We can also migrate easier away from this by just having users remove this function, if we decide for any reason that we want to choose the API scheme building pattern in the future is the way to go. :)